pkg/controller/core.oam.dev/v1alpha2/application/generator.go

```diff
@@ -135,11 +143,25 @@ func convertStepProperties(step *v1beta1.WorkflowStep, app *v1beta1.Application)
 			step.Properties = util.Object2RawExtension(c)
 			return nil
 		}
-
 	}
 	return errors.Errorf("component %s not found", o.Component)
 }
 
+func checkDependsOnValidComponent(dependsOnComponentNames, allComponentNames []string) (string, bool) {
+	if dependsOnComponentNames == nil {
+		return "", true
+	}
+	for _, dc := range dependsOnComponentNames {
+		for _, c := range allComponentNames {
+			if c == dc {
+				return "", true
+			}
+		}
+		return dc, false
```

Comment:
Return here will skip the rest `dependsOnComponentNames` here.

Reply:
> Return here will skip the rest `dependsOnComponentNames` here.

Yes.
I think this check should be one pre-check before deploy the Application. If any one invalid dependency is found ,just break pre-check and report error message .

Reply:
> Return here will skip the rest `dependsOnComponentNames` here.

Thx.
Fixed, and add one test case for this scenario.